src/Drawable.js

```diff
@@ -122,140 +82,47 @@ class Drawable {
-     * @param {number=} optCostumeResolution Optionally, a resolution for the skin.
-     * @private
+     * @returns {[number,number]} the current scaling percentages applied to this Drawable. [100,100] is normal size.
      */
-    _setSkinBitmap (skinMd5ext, optCostumeResolution) {
-        const url = skinMd5ext;
-        this._setSkinCore(url, optCostumeResolution);
-    }
-
-    /**
-     * Load an SVG-based skin. This still needs quite a bit of work to match the
-     * level of quality found in Scratch 2.0:
-     * - We should detect when a skin is being scaled up and render the SVG at a
-     *   higher resolution in those cases.
-     * - Colors seem a little off. This may be browser-specific.
-     * - This method works in Chrome, Firefox, Safari, and Edge but causes a
-     *   security error in IE.
-     * @param {string} skinMd5ext The MD5 and file extension of the SVG skin.
-     * @private
-     */
-    _setSkinSVG (skinMd5ext) {
-        const url = skinMd5ext;
-
-        const svgCanvas = document.createElement('canvas');
-        const svgRenderer = new SvgRenderer(svgCanvas);
-
-        const gotSVG = (err, response, body) => {
-            if (!err) {
-                svgRenderer.fromString(body, () => {
-                    this._setSkinCore(svgCanvas, svgRenderer.getDrawRatio());
-                });
-            }
-        };
-        xhr.get({
-            useXDR: true,
-            url: url
-        }, gotSVG);
-        // TODO: if there's no current u_skin, install *something* before returning
+    get scale () {
+        return [this._scale[0], this._scale[1]];
     }
 
     /**
-     * Common code for setting all skin types.
-     * @param {string|Image} source The source of image data for the skin.
-     * @param {int} costumeResolution The resolution to use for this skin.
-     * @private
+     * @returns {int} A bitmask identifying which effects are currently in use.
      */
-    _setSkinCore (source, costumeResolution) {
-        const callback = (err, texture, sourceInCallback) => {
-            if (!err && (this._pendingSkin === texture)) {
-                this._useSkin(texture, sourceInCallback.width, sourceInCallback.height, costumeResolution);
-            }
-        };
-
-        const gl = this._gl;
-        const options = {
-            auto: true,
-            mag: gl.NEAREST,
-            min: gl.NEAREST, // TODO: mipmaps, linear (except pixelate)
-            wrap: gl.CLAMP_TO_EDGE,
-            src: source
-        };
-        const willCallCallback = typeof source === 'string';
-        this._pendingSkin = twgl.createTexture(gl, options, willCallCallback ? callback : null);
-
-        // If we won't get a callback, start using the skin immediately.
-        // This will happen if the data is already local.
-        if (!willCallCallback) {
-            callback(null, this._pendingSkin, source);
-        }
+    getEnabledEffects () {
```

Comment:
Yeah... there are quite a few things that could and maybe should be converted to getters in here. This particular one didn't actually change in this PR: it just looks like it changed because of the other stuff that moved around near it. I'm planning to address things like this in a post-pen cleanup.